src/engine/sync/enginesync.cpp

```diff
@@ -381,9 +379,30 @@ void EngineSync::notifyPlayingAudible(Syncable* pSyncable, bool playingAudible)
 }
 
 void EngineSync::notifyScratching(Syncable* pSyncable, bool scratching) {
-    // No special behavior for now.
-    Q_UNUSED(pSyncable);
-    Q_UNUSED(scratching);
+    if (!pSyncable->isPlaying() || !pSyncable->isQuantized()) {
+        return;
+    }
+    // Only take action if scratching is turning off.
+    if (scratching) {
+        return;
+    }
+    if (isFollower(pSyncable->getSyncMode())) {
+        pSyncable->getChannel()->getEngineBuffer()->requestSyncPhase();
+        return;
+    }
+    if (isLeader(pSyncable->getSyncMode())) {
```

Comment:
You can invert this condition and exit early to reduce nesting depth.

Reply:
done, and added a synchronized check at the top to simplify further